src/pages/ContactUsView.jsx

```diff
@@ -1,21 +1,90 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import FeatureCopy from "../components/FeatureContainer/FeatureCopy";
 import OlContactBean from '../assets/olLilBean/OlContactBean';
 import Input from '../components/Objects/Input/Input';
 import TextArea from '../components/Objects/TextArea/TextArea';
-import Button from '../components/Objects/Button/Button';
 
   /************************************
    * Constants
    ************************************/
 
-  const BODY_TEXT = 'body';
-  const MESSAGE_TEXT = 'Message';
-  const SUBJECT_TEXT = 'Subject';
   const TARGET_EMAIL = 'mailto:oneleifdev@gmail.com';
 
 export default function ContactUsView() {
+  /************************************
+  * State
+  ************************************/
+
+  const [mailTo, setMailTo] = useState('');
+  const [message, setMessage] = useState('');
+  const [subject, setSubject] = useState('');
+
+  /************************************
+  * Private Methods
+  ************************************/
+
+  /* Takes in the value change event and sets the subject;
+   * Passes the event and other input into a function to be handled
+   *
+   * @param subjectEvent - value change event from the subject input
+   */
+  function handleSubjectInput(subjectEvent) {
+    setSubject(subjectEvent.target.value);
+    handleEnteredInput(subjectEvent, message);
+  } 
+
+  /* Takes in the value change event and sets the message;
+   * Passes the event and other input into a function to be handled
+   *
+   * @param messageEvent - value change event from the message textarea
+   */
+  function handleMessageInput(messageEvent) {
+    setMessage(messageEvent.target.value);
+    handleEnteredInput(messageEvent, subject);
+  } 
```

Comment:
So then i would be doing a check to set which ever value came through, based on what id comes in?

Reply:
Exactly. Something like the following: 

```javascript
const [formData, setFormData] = useState()

function handleInput(event) {
   const {id, value} = event.target;
   setFormData({...formData, [id]: value});

   // Other random stuff
}
```
The only issue is that it is less obvious, but I think it might be worth it.

Reply:
ahhh i see what you mean, way more simple than what i was gonna implement